autonomic-administration-plugin/src/test/java/br/com/autonomiccs/autonomic/administration/plugin/HypervisorManagerTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * This program is part of Autonomiccs "autonomic-platform",
+ * an open source autonomic cloud computing management platform.
+ * Copyright (C) 2016 Autonomiccs, Inc.
+ *
+ * Licensed to the Autonomiccs, Inc. under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The Autonomiccs, Inc. licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package br.com.autonomiccs.autonomic.administration.plugin;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.cloud.host.Host;
+import com.cloud.host.HostVO;
+import com.cloud.hypervisor.Hypervisor.HypervisorType;
+import com.cloud.utils.exception.CloudRuntimeException;
+
+import br.com.autonomiccs.autonomic.administration.plugin.hypervisors.HypervisorHost;
+import br.com.autonomiccs.autonomic.administration.plugin.hypervisors.xenserver.XenHypervisor;
+import br.com.autonomiccs.autonomic.plugin.common.services.HostService;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HypervisorManagerTest {
+
+    @Spy
+    @InjectMocks
+    private HypervisorManager hypervisorManager;
+    @Mock
+    private HostService hostService;
+    @Mock
+    private List<HypervisorHost> hypervisorHosts;
+
+    private HypervisorHost hypervisor;
+
+    @Before
+    public void setup() {
+        hypervisor = Mockito.mock(XenHypervisor.class);
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void shutdownHostTestFull() {
```

Comment:
There is the flow of execution when an exception happens that has to be tested too here.